components/atom/validationCode/src/index.js

```diff
@@ -0,0 +1,65 @@
+import PropTypes from 'prop-types'
+import {useState, useRef, createRef} from 'react'
+
+const DEFAULT_LENGTH = 6
+const BASE_CLASS = 'sui-AtomValidationCode'
+const INPUT_MAX_NUMBER = 10
+const noop = () => {}
+
+export default function AtomValidationCode({
+  onChange = noop,
+  length = DEFAULT_LENGTH
+}) {
+  const [numbers, setNumbers] = useState(new Array(length).fill(null))
+
+  const numberRefs = useRef([])
+  numberRefs.current = numbers.map(
+    (_, i) => numberRefs.current[i] ?? createRef()
+  )
+
+  const handleChange = ({i, value}) => {
+    if (value < INPUT_MAX_NUMBER) {
+      const numberArray = [...numbers]
+      numberArray[i] = value
+      setNumbers(numberArray)
+      onChange(numberArray.join(''))
```

Comment:
event handlers must return as an argument, the raw event and an object with the desired returning values:
```js
onChange(ev, {value: numberArray.join('')})
```